src/components/data_submission/consent_group/EditConsentGroup.js

```diff
@@ -0,0 +1,385 @@
+      id: idx+'_gsText',
+      name: 'gs',
+      validators: [FormValidators.REQUIRED],
+      placeholder: 'Specify Geographic Restriction',
+      defaultValue: gsText,
+      onChange: ({key, value, isValid}) => {
+        setGSText(value);
+        onChange({key: key, value: value, isValid: isValid});
+      },
+    }),
+
+    h(FormField, {
+      id: idx+'_mor',
+      name: 'mor',
+      type: FormFieldTypes.CHECKBOX,
+      toggleText: 'Publication Moratorium (MOR)',
+      defaultValue: showMORText,
+      onChange: ({key, value}) => {
+        setShowMORText(value);
+        onChange({key: key, value: (value ? morText : undefined)});
+      }
+    }),
+
+    h(FormField, {
+      isRendered: showMORText,
+      id: idx+'_morText',
+      name: 'mor',
+      validators: [FormValidators.REQUIRED, FormValidators.DATE],
+      placeholder: 'Please specify date (YYYY-MM-DD)',
+      defaultValue: morText,
+      onChange: ({key, value, isValid}) => {
+        setMORText(value);
+        onChange({key: key, value: value, isValid: isValid});
+      },
+    }),
+
+    h(FormField, {
+      id: idx+'_npu',
+      name: 'npu',
+      type: FormFieldTypes.CHECKBOX,
+      toggleText: 'Non-profit Use Only (NPU)',
+      defaultValue: consentGroup.npu,
+      onChange
+    }),
+
+    h(FormField, {
+      id: idx+'_otherSecondary',
+      name: 'otherSecondary',
+      type: FormFieldTypes.CHECKBOX,
+      toggleText: 'Other',
+      defaultValue: showOtherSecondaryText,
+      onChange: ({key, value}) => {
+        setShowOtherSecondaryText(value);
+
+        if (value) {
+          onChange({key: key, value: otherSecondaryText});
+        } else {
+          onChange({key: key, value: undefined});
+        }
+      }
+    }),
+
+    h(FormField, {
+      isRendered: showOtherSecondaryText,
+      id: idx+'_otherSecondaryText',
+      name: 'otherSecondary',
+      validators: [FormValidators.REQUIRED],
+      placeholder: 'Please specify',
+      defaultValue: otherSecondaryText,
+      onChange: ({key, value, isValid}) => {
+        setOtherSecondaryText(value);
+        onChange({key: key, value: value, isValid: isValid});
+      },
+    }),
+
+    // location
+
+    h(FormField, {
+      id: idx+'_dataLocation',
+      name: 'dataLocation',
+      type: FormFieldTypes.SELECT,
+      isMulti: true,
+      title: 'Data Location',
+      description: 'Please provide the location of your data resource for this consent group',
+      exclusiveValues: ['Not Determined'],
+      selectOptions: [
+        'AnVIL Workspace',
+        'Terra Workspace',
+        'TDR Location',
+        'Not Determined',
+      ],
+      defaultValue: consentGroup.dataLocation,
+      placeholder: 'Data Location(s)',
+      validators: [FormValidators.REQUIRED],
+      onChange,
+    }),
+
+    h(FormField, {
+      id: idx+'_url',
+      name: 'url',
+      title: 'Data URL',
+      validators: [FormValidators.REQUIRED, FormValidators.URL],
```

Comment:
Minor issue on the URL Form validator. `www.` isn't always representative of a valid url. For example, `www.duos.org` or `https://www.duos.org` are invalid urls, but `duos.org` or `https://duos.org` will get us to the DUOS production page.

Reply:
Wait so what is the desired behavior here?

`duos.org` should pass, and `www.duos.org` should not?

Reply:
No, `www.` shouldn't be a requirement at all. You can try this out with the DUOS example I listed. Typing in `www.duos.org` will give you a 'Site can't be reached' but `duos.org` will get you to our production site.

`www` is not a hard requirement for websites. It's up to the domain administrator to determine whether or not they want their website accessible by pre-prending `www` or not.

Reply:
`https://duos.org` should work; I am currently using javascript's inbuilt `new URL()` functionality to test this. I believe the only catch is that you have to specify `http://` or `https://` for it to parse properly. If we want `duos.org` to pass I can change the functionality to assume `https://` schema always.

Reply:
It's not a matter of functionality, its the issue of User Experience. 

The average user does not type in `https://www.google.com`, they type in `google.com`. From there the browser executes the request under `https` for you, and the DNS on google's servers will redirect you to `www.google.com`.

Now imagine I'm a researcher for DUOS. I type in `duos.org` because that's what I always do.

<img width="1571" alt="Screen Shot 2022-09-23 at 8 18 57 AM" src="https://user-images.githubusercontent.com/6664778/191958634-ca3a511d-2283-4f89-b5cd-8cfd50b728da.png">

Validator is now telling me that the url is invalid and provides me `https://www.google.com` as a valid url. Me, the user, will then input `https://www.duos.org` to match the example given.

<img width="1574" alt="Screen Shot 2022-09-23 at 8 20 45 AM" src="https://user-images.githubusercontent.com/6664778/191958993-816ad73e-46b9-47d4-a8e2-1e536c040a80.png">

Great, the validator isn't complaining anymore. The problem is the URL is wrong and any user who views that url on the UI will copy and paste that link on their browser only to find themselves at a page that does not exist.

The application essentially prompted the user to input an invalid URL in the field, meaning we're potentially guiding users to input incorrect information.

Reply:
so, would be assuming `https://` always and showing `duos.org` as an example url be enough to fix the user experience?

Reply:
URL validation now is simpler and allows `duos.org`. Provides `duos.org` as an example.

Reply:
Yep this works. This way we're allowing the user to input the url without influencing them in anyway. If the value ends up being wrong we can either point people to SOs or contact them (SOs) directly and tell them to fix it.